src/smartpeak/source/core/RawDataProcessor.cpp

```diff
@@ -2413,4 +2423,231 @@ namespace SmartPeak
 
     LOGD << "END CalculateMDVAccuracies";
   }
+
+  ParameterSet PickMS2Features::getParameterSchema() const
+  {
+    OpenMS::MassTraceDetection mass_trace_detection;
+    OpenMS::ElutionPeakDetection elution_peak_detection;
+    OpenMS::FeatureFindingMetabo feature_finding_metabo;
+    ParameterSet parameters({ mass_trace_detection, elution_peak_detection, feature_finding_metabo });
+    std::map<std::string, std::vector<std::map<std::string, std::string>>> param_struct({
+    {"PickMS2Features", {
+    {
+      {"name", "enable_elution"},
+      {"type", "bool"},
+      {"value", "true"},
+      {"description", "Enable elution peak detection"},
+    },
+    {
+      {"name", "force_processing"},
+      {"type", "bool"},
+      {"value", "false"},
+      {"description", "To enforce processing of the data when Profile data is provided but spectra is not centroided"},
+    },
+    {
+      {"name", "max_traces"},
+      {"type", "int"},
+      {"value", "0"},
+      {"description", "Max traces to use for MassTraceDetection. 0 means all traces."},
+    }
+    }} });
+    ParameterSet pick_ms2_feature_params(param_struct);
+    parameters.merge(pick_ms2_feature_params);
+    return parameters;
+  }
+
+  void PickMS2Features::process(
+    RawDataHandler& rawDataHandler_IO,
+    const ParameterSet& params_I,
+    const Filenames& filenames
+  ) const
+  {
+    LOGD << "START PickMS2Features";
+
+    //-------------------------------------------------------------
+    // set parameters
+    //-------------------------------------------------------------
+    FunctionParameters pick_ms2_feature_params;
+    if (params_I.count("PickMS2Features"))
+    {
+      pick_ms2_feature_params = params_I.at("PickMS2Features");
+    }
+
+    OpenMS::MassTraceDetection mtdet;
+    OpenMS::Param mtdet_parameters = mtdet.getParameters();
+    mtdet_parameters.remove("chrom_fwhm");
+    Utilities::updateParameters(mtdet_parameters, params_I.at("MassTraceDetection"));
+
+    OpenMS::ElutionPeakDetection epdet;
+    OpenMS::Param epdet_parameters = epdet.getParameters();
+    Utilities::updateParameters(epdet_parameters, params_I.at("ElutionPeakDetection"));
+
+    OpenMS::FeatureFindingMetabo ffmet;
+    OpenMS::Param ffmet_parameters = ffmet.getParameters();
+    Utilities::updateParameters(ffmet_parameters, params_I.at("MRMFeatureFinderScoring"));
+
+    //-------------------------------------------------------------
+    // Setting input
+    //-------------------------------------------------------------
+    OpenMS::PeakMap ms_peakmap;
+    std::vector<int> ms_level(1, 1);
+    for (OpenMS::MSSpectrum& spec : rawDataHandler_IO.getExperiment().getSpectra()) {
+      ms_peakmap.addSpectrum(spec);
```

Comment:
Given the comment above about wanting to extract out only MS1 level features, we will want to have a check in hear that queries the ms level (i.e., `getMSLevel()`) and only adds in spectra that are MS level 1.  The spectra should be a mix of both MS level 1 and 2.

Reply:
Please let me know if you need any additional clarifications on this.